ddht/rpc_handlers.py

```diff
@@ -84,9 +87,59 @@ async def do_call(self, params: None) -> NodeInfoResponse:
         )
 
 
+@to_tuple
+def normalize_and_validate_kv_pairs(
+    params: Any,
+) -> Iterable[Tuple[bytes, Optional[bytes]]]:
+    if not params:
+        raise RPCError("Missing parameters.")
+
+    for kv_pair in params:
+        if len(kv_pair) != 2:
+            raise RPCError(f"Invalid kv_pair length: {len(kv_pair)}.")
+
+        if not is_text(kv_pair[0]):
```

Comment:
I don't think we're going to be able to do everything with unicode encoded bytes since you cannot actually represent the full range of bytes in unicode.  I'm thinking we might want to do:

- key
    - is_text -> must be hexstr
- value
    - is_text -> must be hexstr
    - is_integer -> treat as integer
    - is_none -> `None`